{"metadata":{"title":"getRetryWaitTime(error:attempt:)","roleHeading":"Instance Method","modules":[{"name":"SotoCore"}],"symbolKind":"method","required":true,"role":"symbol","externalID":"s:8SotoCore11RetryPolicyP03getC8WaitTime5error7attemptAA0C6StatusOSgs5Error_p_SitF","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"getRetryWaitTime"},{"kind":"text","text":"("},{"kind":"externalParam","text":"error"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":", "},{"kind":"externalParam","text":"attempt"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Int","preciseIdentifier":"s:Si"},{"kind":"text","text":") -> "},{"kind":"typeIdentifier","text":"RetryStatus","preciseIdentifier":"s:8SotoCore11RetryStatusO"},{"kind":"text","text":"?"}]},"schemaVersion":{"major":0,"patch":0,"minor":3},"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy\/getRetryWaitTime(error:attempt:)"},"sections":[],"hierarchy":{"paths":[["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy"]]},"primaryContentSections":[{"declarations":[{"platforms":["Linux"],"tokens":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"getRetryWaitTime","kind":"identifier"},{"text":"(","kind":"text"},{"text":"error","kind":"externalParam"},{"text":": ","kind":"text"},{"text":"Error","kind":"typeIdentifier","preciseIdentifier":"s:s5ErrorP"},{"text":", ","kind":"text"},{"text":"attempt","kind":"externalParam"},{"text":": ","kind":"text"},{"text":"Int","kind":"typeIdentifier","preciseIdentifier":"s:Si"},{"text":") -> ","kind":"text"},{"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryStatus","text":"RetryStatus","kind":"typeIdentifier","preciseIdentifier":"s:8SotoCore11RetryStatusO"},{"text":"?","kind":"text"}],"languages":["swift"]}],"kind":"declarations"},{"parameters":[{"content":[{"type":"paragraph","inlineContent":[{"type":"text","text":"Error returned by HTTP client"}]}],"name":"error"},{"content":[{"inlineContent":[{"text":"Retry attempt number","type":"text"}],"type":"paragraph"}],"name":"attempt"}],"kind":"parameters"}],"abstract":[{"text":"Returns whether we should retry and how long we should wait before retrying","type":"text"}],"kind":"symbol","variants":[{"paths":["\/documentation\/sotocore\/retrypolicy\/getretrywaittime(error:attempt:)"],"traits":[{"interfaceLanguage":"swift"}]}],"references":{"doc://codes.soto.soto-core/documentation/SotoCore":{"abstract":[{"type":"text","text":"The core framework for Soto, a Swift SDK for AWS"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","title":"SotoCore","role":"collection","type":"topic","url":"\/documentation\/sotocore","kind":"symbol"},"doc://codes.soto.soto-core/documentation/SotoCore/RetryStatus":{"navigatorTitle":[{"text":"RetryStatus","kind":"identifier"}],"url":"\/documentation\/sotocore\/retrystatus","type":"topic","kind":"symbol","title":"RetryStatus","role":"symbol","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"RetryStatus","kind":"identifier"}],"abstract":[{"text":"Return value for ","type":"text"},{"type":"codeVoice","code":"RetryPolicy.getRetryWaitTime"},{"text":". Either retry after time amount or donâ€™t retry","type":"text"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryStatus"},"doc://codes.soto.soto-core/documentation/SotoCore/RetryPolicy/getRetryWaitTime(error:attempt:)":{"title":"getRetryWaitTime(error:attempt:)","type":"topic","role":"symbol","required":true,"abstract":[{"type":"text","text":"Returns whether we should retry and how long we should wait before retrying"}],"kind":"symbol","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"getRetryWaitTime","kind":"identifier"},{"text":"(","kind":"text"},{"text":"error","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:s5ErrorP","text":"Error","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"text":"attempt","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:Si","text":"Int","kind":"typeIdentifier"},{"text":") -> ","kind":"text"},{"preciseIdentifier":"s:8SotoCore11RetryStatusO","text":"RetryStatus","kind":"typeIdentifier"},{"text":"?","kind":"text"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy\/getRetryWaitTime(error:attempt:)","url":"\/documentation\/sotocore\/retrypolicy\/getretrywaittime(error:attempt:)"},"doc://codes.soto.soto-core/documentation/SotoCore/RetryPolicy":{"abstract":[{"text":"Protocol for Retry strategy. Has function returning amount of time before the next retry after an HTTP error","type":"text"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy","navigatorTitle":[{"kind":"identifier","text":"RetryPolicy"}],"title":"RetryPolicy","role":"symbol","fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"RetryPolicy"}],"type":"topic","url":"\/documentation\/sotocore\/retrypolicy","kind":"symbol"}}}